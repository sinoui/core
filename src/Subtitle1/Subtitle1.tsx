import { TextContent, typographyPub } from '../utils/typographyPub';

const Subtitle1 = typographyPub('h6', 'subtitle1')(TextContent);
export default Subtitle1;
