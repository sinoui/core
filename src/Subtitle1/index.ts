import Subtitle1 from './Subtitle1';

export default Subtitle1;
