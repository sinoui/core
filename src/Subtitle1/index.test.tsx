import React from 'react';
import renderer from 'react-test-renderer';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from '@sinoui/theme';
import Subtitle1 from './index';

/**
 * Subtitle1组件单元测试
 */
describe('文字排版 单元测试', () => {
  afterEach(cleanup);

  it('指定不同的根元素', () => {
    const { container } = render(
      <ThemeProvider theme={defaultTheme}>
        <Subtitle1 as="div">这是一个文字。</Subtitle1>
      </ThemeProvider>,
    );

    const text = container.querySelector('div');
    expect(text).toHaveTextContent('这是一个文字。');
  });
  it('通过 gutterBottom 指定下边距', () => {
    const { container } = render(
      <ThemeProvider theme={defaultTheme}>
        <Subtitle1 gutterBottom>这是一个文字。</Subtitle1>
      </ThemeProvider>,
    );

    const text = container.querySelector('.sinoui-typography--gutter-bottom');
    expect(text).toHaveTextContent('这是一个文字。');
  });
  it('ref 指向根元素', () => {
    const ref: React.RefObject<HTMLInputElement> = React.createRef();
    const { container } = render(
      <ThemeProvider theme={defaultTheme}>
        <Subtitle1 data-testid="typo" ref={ref}>
          这是一个文字。
        </Subtitle1>
      </ThemeProvider>,
    );

    const text = container.firstChild;
    expect(text).toEqual(ref.current);
  });
  it('设置align文本对齐方式', () => {
    const { container } = render(
      <ThemeProvider theme={defaultTheme}>
        <Subtitle1 data-testid="typo" align="center">
          这是一个文字。
        </Subtitle1>
      </ThemeProvider>,
    );

    const text = container.firstChild;
    expect(text).toHaveStyle(`text-align:center`);
  });
});

/**
 * Subtitle1组件快照测试
 */
describe('文字排版 快照测试', () => {
  it('基本文字排版', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <Subtitle1>Subtitle1</Subtitle1>
        </ThemeProvider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('设置属性', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <Subtitle1 as="h2">Subtitle1</Subtitle1>
          <Subtitle1 align="center">Subtitle1</Subtitle1>
          <Subtitle1 noWrap>Subtitle1</Subtitle1>
          <Subtitle1 color="success">Subtitle1</Subtitle1>
          <Subtitle1 gutterBottom>Subtitle1</Subtitle1>
        </ThemeProvider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('颜色设置', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <Subtitle1>Subtitle1</Subtitle1>
          <Subtitle1 color="primary">Subtitle1</Subtitle1>
          <Subtitle1 color="secondary">Subtitle1</Subtitle1>
          <Subtitle1 color="textPrimary">Subtitle1</Subtitle1>
          <Subtitle1 color="textSecondary">Subtitle1</Subtitle1>
          <Subtitle1 color="error">Subtitle1</Subtitle1>
          <Subtitle1 color="warning">Subtitle1</Subtitle1>
          <Subtitle1 color="success">Subtitle1</Subtitle1>
          <Subtitle1 color="info">Subtitle1</Subtitle1>
        </ThemeProvider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
