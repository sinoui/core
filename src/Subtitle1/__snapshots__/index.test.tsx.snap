// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`文字排版 快照测试 基本文字排版 1`] = `
<h6
  className="sc-bdnylx fAHZBl sinoui-typography sinoui-typography--subtitle1"
  color="textPrimary"
  type="subtitle1"
>
  Subtitle1
</h6>
`;

exports[`文字排版 快照测试 设置属性 1`] = `
Array [
  <h2
    className="sc-bdnylx fAHZBl sinoui-typography sinoui-typography--subtitle1"
    color="textPrimary"
    type="subtitle1"
  >
    Subtitle1
  </h2>,
  <h6
    className="sc-bdnylx eJWFhl sinoui-typography sinoui-typography--subtitle1"
    color="textPrimary"
    type="subtitle1"
  >
    Subtitle1
  </h6>,
  <h6
    className="sc-bdnylx fhnHKI sinoui-typography sinoui-typography--subtitle1"
    color="textPrimary"
    type="subtitle1"
  >
    Subtitle1
  </h6>,
  <h6
    className="sc-bdnylx ghgSjh sinoui-typography sinoui-typography--subtitle1"
    color="success"
    type="subtitle1"
  >
    Subtitle1
  </h6>,
  <h6
    className="sc-bdnylx lhkpDt sinoui-typography sinoui-typography--subtitle1 sinoui-typography--gutter-bottom"
    color="textPrimary"
    type="subtitle1"
  >
    Subtitle1
  </h6>,
]
`;

exports[`文字排版 快照测试 颜色设置 1`] = `
Array [
  <h6
    className="sc-bdnylx fAHZBl sinoui-typography sinoui-typography--subtitle1"
    color="textPrimary"
    type="subtitle1"
  >
    Subtitle1
  </h6>,
  <h6
    className="sc-bdnylx dIBgPu sinoui-typography sinoui-typography--subtitle1"
    color="primary"
    type="subtitle1"
  >
    Subtitle1
  </h6>,
  <h6
    className="sc-bdnylx cCQgoh sinoui-typography sinoui-typography--subtitle1"
    color="secondary"
    type="subtitle1"
  >
    Subtitle1
  </h6>,
  <h6
    className="sc-bdnylx fAHZBl sinoui-typography sinoui-typography--subtitle1"
    color="textPrimary"
    type="subtitle1"
  >
    Subtitle1
  </h6>,
  <h6
    className="sc-bdnylx dOAKmE sinoui-typography sinoui-typography--subtitle1"
    color="textSecondary"
    type="subtitle1"
  >
    Subtitle1
  </h6>,
  <h6
    className="sc-bdnylx kzZnhc sinoui-typography sinoui-typography--subtitle1"
    color="error"
    type="subtitle1"
  >
    Subtitle1
  </h6>,
  <h6
    className="sc-bdnylx fJoYBt sinoui-typography sinoui-typography--subtitle1"
    color="warning"
    type="subtitle1"
  >
    Subtitle1
  </h6>,
  <h6
    className="sc-bdnylx ghgSjh sinoui-typography sinoui-typography--subtitle1"
    color="success"
    type="subtitle1"
  >
    Subtitle1
  </h6>,
  <h6
    className="sc-bdnylx ioUlTJ sinoui-typography sinoui-typography--subtitle1"
    color="info"
    type="subtitle1"
  >
    Subtitle1
  </h6>,
]
`;
